autoload/test/zig/zigtest.vim

```diff
@@ -8,8 +8,12 @@ endfunction
 
 function! test#zig#zigtest#build_position(type, position) abort
   if a:type ==# 'nearest'
-    " --test-filter currently only supports filtering string inclusion, so
-    "  multiple tests could match, so instead we run the whole file.
+    let name = s:nearest_test(a:position)
+    if !empty(name)
+      let filter = '--test-filter '.shellescape(name, 1)
+      return ['test', a:position['file'], filter]
+    endif
+
     return ['test', a:position['file']]
```

Comment:
This could be refactored to:

```suggestion
    if empty(name)
      return ['test', a:position['file']]
    endif

    return ['test', a:position['file'],  '--test-filter '.shellescape(name, 1)]
```